src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -219,28 +228,109 @@ ExpressionSyntax initializer
 				immutable = false;
 			}
 
-			var stuff = GetStuffToCheckForMember(
-				type,
-				typeSyntax,
-				initializer
-			);
+			// Check all assignments of the member for immutability
+			foreach( var assignment in assignments ) {
+				var stuff = GetStuffToCheckForMember(
+					type,
+					typeSyntax,
+					assignment
+				);
 
-			if ( stuff == null ) {
-				// null is a signal that there is nothing further that needs to
-				// be checked
-				return immutable;
-			}
+				if( stuff == null ) {
+					// null is a signal that there is nothing further that needs to
+					// be checked
+					continue;
+				}
+
+				var (typeToCheck, checkKind, getLocation) = stuff.Value;
 
-			var (typeToCheck, checkKind, getLocation) = stuff.Value;
+				if( m_context.IsImmutable(
+					typeToCheck,
+					checkKind,
+					getLocation,
+					out var diagnostic
+				) ) {
+					continue;
+				}
 
-			if( !m_context.IsImmutable( typeToCheck, checkKind, getLocation, out var diagnostic ) ) {
 				diagnosticSink( diagnostic );
 				immutable = false;
 			}
 
 			return immutable;
 		}
 
+
+		/// <summary>
+		/// A readonly field or property may be re-assigned in constructors.
+		/// This method finds all assignments of the field/property.
+		/// </summary>
+		/// <param name="symbol">The field/property to search for</param>
+		/// <param name="initializer">The initialization with the declaration</param>
+		/// <returns>An immutable list of assignments</returns>
+		private ImmutableArray<ExpressionSyntax> GetAssignments(
+			ISymbol symbol,
+			ExpressionSyntax initializer
+		) {
+			List<ExpressionSyntax> assignments = new() {initializer};
+
+			// If the containing symbol isn't a named type (class) then there
+			// will be no constructors
+			if( symbol.ContainingSymbol is not INamedTypeSymbol namedTypeSymbol ) {
+				return assignments.ToImmutableArray();
+			}
+
+			foreach( var methodSymbol in namedTypeSymbol.Constructors ) {

```

Comment:
It got lost in the shuffle, but: https://github.com/Brightspace/D2L.CodeStyle/pull/698#discussion_r566178258